aptos-move/block-executor/src/executor.rs

```diff
@@ -644,8 +646,8 @@ where
         let executor = E::init(executor_arguments);
         drop(init_timer);
 
+        let counter = Rc::new(RefCell::new(0));
```

Comment:
I'm actually confused with the counter. I thought there would be one shared counter that will be used by everyone. But there is a counter here in executor.rs, there is another counter in executor_view_base.rs, there is another counter in ParallelState/SequentialState, etc.

Reply:
Just to make sure I understood correctly - this counter is for when we are doing sequential execution, and the other one is for when we are doing parallel execution? 
Is ExecutorViewBase not used in sequential execution?


and no matter which execution is used, there is only a single counter in existance, correct?

Reply:
Yes, so we need to support
1. Parallel execution
2. Sequential execution
3. Simulating txns, e.g. running view functions, etc. or tests without block-stm.

For (1) we use an atomic which is passed to ParallelState, for (2) we use this and pass to `Unsync` map, for (3) I added a counter to the adapter which makes the state view be treated as executor view, to be able to simulate txns, etc. So it is always 1 counter in existence.